app/src/main/java/com/example/grutor/Activites/RegisterActivity.java

```diff
@@ -58,20 +59,21 @@ public void onClick(View v) {
 
             public void registerUser() {
                 // Create the ParseUser
-                user = new ParseUser();
+               // user = new ParseUser();
                 // Set core properties
                 Log.i(TAG, "Inside the registerUser function.");
                 String email = etUserLogin.getText().toString();
                 String password = etPassword.getText().toString();
                 String firstName = etFirstName.getText().toString();
+                String fullName = etFirstName.getText().toString() + " " + etLastName.getText().toString();
 
```

Comment:
you should check the format of email. the length of user name and password. [Optional] check if any duplicated user name too( or show a error message when user sign up a duplicated user name).

Reply:
I have implemented a new email checker function and the changes about adding and extracting String resources from strings.xml. I will go back to checking the length of the user name and password later. I added a todo, thanks!